### ISSUE TITLE
Unable to pass paratmers to file in dotfile install command

### ISSUE THREAD

**Issue by ankitbko** [microsoft/vscode-remote-release#3047]
- VSCode Version: 
- Local OS Version:
- Remote OS Version:
For all the above - 
Version: 1.45.1 (user setup)
Commit: 5763d909d5f12fe19f215cbfdd29a91c0fa9208a
Date: 2020-05-14T08:27:35.169Z
Electron: 7.2.4
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Windows_NT x64 10.0.19041
- Remote Extension/Connection Type: SSH/Docker/WSL: docker (devcontainer)

Steps to Reproduce:

1. In dotfile install command pass parameter to the file like `~/dotfiles/install.sh force`
2.  Dotfile execution fails as it says the file is not executable
```
[9987 ms] Start: Run in container: # Clone & install dotfiles
[11573 ms] ~/dotfiles/install.sh force not executable
[11573 ms] Cloning into '/home/vscode/dotfiles'...
/bin/sh: 39: [: ~/dotfiles/install.sh: unexpected operator
```

The issue seems to be with this line in [dotfiles.ts](https://github.com/microsoft/vscode-remote-containers/blob/651cc996bfb5303a8d5618ba248007d876da84ec/src/node/dotfiles.ts#L29). I don't know if this is fixed in the latest code, I can't find the same file in master branch.

`-x filepah` returns true only if the `filepath` exists and is executable. I couldn't make it work with `filepath` having the arguments to file. I have no idea if there is some shell magic to make it work.

**Comment by skycaptain** [comment]
Run into the same issue. I'm trying to use [chezmoi](https://github.com/twpayne/chezmoi), where the install command is `chezmoi apply`. Any news?

**Comment by skycaptain** [comment]
Ok, the solution was quite easy: just add an entry script to your dotfiles repo. Actually, I like this even more, since users might use different dotfiles managers and this way I'm able to install chezmoi only for me via the entry script as opposed to install it in the project's Dockerfile definition:

```shell
#!/usr/bin/env bash

# Exit on error
set -o errexit
# Exit on error inside any functions or subshells
set -o errtrace

# Install chezmoi if not installed
if ! [ -x "$(command -v chezmoi)" ]; then
    sudo BINDIR=/usr/local/bin sh -c "$(wget -O- https://git.io/chezmoi)"
fi

# Create dotfiles directory
mkdir -p ~/.local/share/chezmoi

# Install dotfiles
chezmoi apply --force
```

In your settings.json:

```json
{
    ...
    "remote.containers.dotfiles.installCommand": "~/.local/share/chezmoi/install-devcontainer.sh",
    "remote.containers.dotfiles.repository": ...,
    "remote.containers.dotfiles.targetPath": "~/.local/share/chezmoi",
    ...
}
```

**Comment by ankitbko** [comment]
Yes I have the entry script as well. The feature request is to allow arguments to be passed to the entry script. This gives greater flexibility to setup dotfiles as needed for each workspace. The workaround I do now is to have multiple entry script and call the one needed.

**Comment by jrussellsmyth** [comment]
The ability to specify an install command with args would be very helpful for cases where you want different things added/executed/linked per project but a single install entrypoint that handles default scenario. The workaround of having repo-specific entry points works, but is less efficient, and does not allow for default.

**Comment by tudortimi** [comment]
I just ran into the same issue. I was also trying to set up my dotfiles repo managed with `chezmoi`. The command I wanted to use was:

```
sh -c "$(curl -fsLS get.chezmoi.io)" -- init --apply --promptString name="Tudor Timi",email="tudortimi@users.noreply.github.com"
```

This would download the tool without having to touch the Dockerfile. It would then call it and pass it some configuration values. I can put this into a script, but this would mean I'd have to hard-code the configuration values.

Would be nice if args for the command were supported.
